core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.cache;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.exception.PageNotFoundException;
+
+import java.io.IOException;
+import java.nio.channels.ReadableByteChannel;
+
+import javax.annotation.Nullable;
+
+/**
+ * Interface for managing cached pages.
+ */
+public interface CacheManager extends AutoCloseable  {
+  /**
+   * @param conf the Alluxio configuration
+   * @return an instance of {@link CacheManager}
+   */
+  static CacheManager create(AlluxioConfiguration conf) throws IOException {
+    // TODO(feng): make cache manager type configurable when we introduce more implementations.
+    return LocalCacheManager.create(conf);
+  }
+
+  /**
+   * Writes a new page from a source channel with best effort. It is possible that this put
+   * operation returns without page written due to transient behavior not due to failures writing
+   * to disks.
+   *
+   * @param pageId page identifier
+   * @param page page data
+   * @throws IOException if error happens when writing the page to disk
+   * @return true on a successful put or false due to transient
+   */
+  boolean put(PageId pageId, byte[] page) throws IOException;
```

Comment:
Since this is best effort, could make this API not throw exception, even on write failure?